packages/cactus-plugin-satp-hermes/src/main/typescript/core/recovery/rollback/stage0-rollback-strategy.ts

```diff
@@ -0,0 +1,74 @@
+import { Logger, LoggerProvider } from "@hyperledger/cactus-common";
+import { SATPSession } from "../../satp-session";
+import { RollbackState, RollbackStrategy } from "./rollback-strategy-factory";
+import { RollbackLogEntry } from "../../../generated/proto/cacti/satp/v02/crash_recovery_pb";
+
+export class Stage0RollbackStrategy implements RollbackStrategy {
+  private log: Logger;
+  private rollbackLogEntry: RollbackLogEntry;
+
+  constructor(logEntry: RollbackLogEntry) {
+    this.log = LoggerProvider.getOrCreate({ label: "Stage0RollbackStrategy" });
+    this.rollbackLogEntry = logEntry;
+  }
+
+  // return a rollback state in all strategies
+  async execute(session: SATPSession): Promise<RollbackState> {
+    const fnTag = "Stage0RollbackStrategy#execute";
+    this.log.info(`${fnTag} Executing rollback for Stage 0`);
+
+    // check session exists
+    if (!session) {
+      throw new Error(`${fnTag}, session data is not correctly initialized`);
+    }
+    try {
+      // TODO record the rollback on the log. Implement RollbackLogEntry
```

Comment:
Please remove todos when the task is completed. I see you are already recording the rollback log entry (despite being a dummy entry). Please un-hardcode it, and gather information from the current session